doc/lexer_keywords.md

```diff
@@ -0,0 +1,11 @@
+# Introduction
+
+This document outlines how to add new keywords to the lexer:
```

Comment:
We do it all the time as developers, so 1.4 is adding 'GROUP BY' and 1.5+ 'LIMIT', 'ORDER BY', 'DISTINCT' etc, etc